sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionsClient.java

```diff
@@ -0,0 +1,388 @@
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return represents some basic information about the AI model along with {@link Response}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    Response<BinaryData> getModelInfoWithResponse(RequestOptions requestOptions) {
+        return this.serviceClient.getModelInfoWithResponse(requestOptions);
+    }
+
+    /**
+     * Gets chat completions for the provided chat messages.
+     * Completions support a wide variety of tasks and generate text that continues from or "completes"
+     * provided prompt data. The method makes a REST API call to the `/chat/completions` route
+     * on the given endpoint.
+     *
+     * @param options Options for complete API.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return chat completions for the provided chat messages.
+     * Completions support a wide variety of tasks and generate text that continues from or "completes"
+     * provided prompt data.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public ChatCompletions complete(ChatCompletionsOptions options) {
+        // Generated convenience method for completeWithResponse
+        RequestOptions requestOptions = new RequestOptions();
+        CompleteRequest completeRequestObj
+            = new CompleteRequest(options.getMessages()).setFrequencyPenalty(options.getFrequencyPenalty())
+                .setStream(options.isStream())
+                .setPresencePenalty(options.getPresencePenalty())
+                .setTemperature(options.getTemperature())
+                .setTopP(options.getTopP())
+                .setMaxTokens(options.getMaxTokens())
+                .setResponseFormat(options.getResponseFormat())
+                .setStop(options.getStop())
+                .setTools(options.getTools())
+                .setToolChoice(options.getToolChoice())
+                .setSeed(options.getSeed())
+                .setModel(options.getModel());
+        BinaryData completeRequest = BinaryData.fromObject(completeRequestObj);
+        ExtraParameters extraParams = options.getExtraParams();
+        if (extraParams != null) {
+            requestOptions.setHeader(HttpHeaderName.fromString("extra-parameters"), extraParams.toString());
+        }
+        return Objects.requireNonNull(completeWithResponse(completeRequest, requestOptions)).getValue();
```

Comment:
Do we need to `Objects.requireNonNull` check here?

Reply:
[Updated](https://github.com/Azure/azure-sdk-for-java/pull/41482/commits/e16676f12ecb82d477715c3343df29291a46e831)